includes/polling/os/awplus.inc.php

```diff
@@ -9,7 +9,16 @@
 $serial = snmp_getnext($device, "rscBoardSerialNumber", "-OQv", "AT-RESOURCE-MIB");
 
 // SBx8100 platform has line cards show up first in "rscBoardName" above.
+//Instead use sysObjectID.0
 if (strpos($hardware, 'SBx81') !== false) {
     $hardware = snmp_get($device, "sysObjectID.0", "-OQvs", "SNMPv2-MIB:AT-PRODUCT-MIB");
-    $hardware = str_replace('at', '', $hardware);
+    $hardware = str_replace('at', 'AT-', $hardware);
+// Features and Serial is set to Controller card  1.5
+    $features = snmp_get($device, "rscBoardName.5.6", "-OQv", "AT-RESOURCE-MIB");
```

Comment:
No way to check sysObjectID or something we already know to determine the rcsBoardNames?

If not, you'll need to convert the two snmp_gets into snmp_get_multi_oid.

Reply:
The problem is the device names suck in AT-PRODUCT-MIB. 

Unless I'm doing something stupidly wrong, this is what returns when polling several different products:

```librenms@librenmsdev:~$  snmpget -v2c -c librenms -OvQs -M /opt/librenms/mibs/:/opt/librenms/mibs/awplus 10.37.105.1 SNMPv2-MIB::sysObjectID.0
MIB search path: /opt/librenms/mibs/:/opt/librenms/mibs/awplus
atsbx908g2 ----> Should be AT-SBx908-GEN2
librenms@librenmsdev:~$  snmpget -v2c -c librenms -OvQs -M /opt/librenms/mibs/:/opt/librenms/mibs/awplus 10.37.105.4 SNMPv2-MIB::sysObjectID.0
MIB search path: /opt/librenms/mibs/:/opt/librenms/mibs/awplus
atx93028GTX ----> Should be AT-x930-28GTX
librenms@librenmsdev:~$  snmpget -v2c -c librenms -OvQs -M /opt/librenms/mibs/:/opt/librenms/mibs/awplus 10.37.105.5 SNMPv2-MIB::sysObjectID.0
MIB search path: /opt/librenms/mibs/:/opt/librenms/mibs/awplus
atXS916MXT ----> Should be AT-XS916-MXT
librenms@librenmsdev:~$  snmpget -v2c -c librenms -OvQs -M /opt/librenms/mibs/:/opt/librenms/mibs/awplus 10.37.105.6 SNMPv2-MIB::sysObjectID.0
MIB search path: /opt/librenms/mibs/:/opt/librenms/mibs/awplus
Cannot find module (FCMGMT-MIB): At line 28 in /opt/librenms/mibs//SW-MIB
Did not find 'connUnitPortStatEntry' in module #-1 (/opt/librenms/mibs//SW-MIB)
atx510L28GT ----> Should be AT-x510-28GT
librenms@librenmsdev:~$  snmpget -v2c -c librenms -OvQs -M /opt/librenms/mibs/:/opt/librenms/mibs/awplus 10.37.105.7 SNMPv2-MIB::sysObjectID.0
MIB search path: /opt/librenms/mibs/:/opt/librenms/mibs/awplus
atGS948MPX ----> Should be AT-GS948-MPX
```

rscBoardName simply provides the more correct device name. It just so happens the SBx8112 and SBx8106 don't have any "dashes" then product type after the name so this worked OK for using AT-PRODUCT-MIB

Reply:
>If not, you'll need to convert the two snmp_gets into snmp_get_multi_oid.

 And I should know this by now ;-)